server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/DFR3MappingResource.java

```diff
@@ -1,25 +1,82 @@
 package edu.illinois.ncsa.incore.service.project.models;
 
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class DFR3MappingResource extends ProjectResource {
 
-    // Enum for status
     public enum Type {
         fragility,
         restoration,
         repair,
-        recovery
+        recovery,
+        unknown;
+
+        // Map a string to a Type, ignoring case
+        public static Type fromString(String mappingType) {
+            try {
+                return Type.valueOf(mappingType.toLowerCase());
+            } catch (IllegalArgumentException | NullPointerException e) {
+                return unknown;
+            }
+        }
     }
-    public Type type;
+
+    private Type type;
+    private String mappingType;
+
+    public String name;
+    public String hazardType;
+    public String inventoryType;
+
+    public final List<Mapping> mappings = new ArrayList<>();
+    public String creator;
+    public String owner;
+
+    public MappingEntryKey[] mappingEntryKeys;
 
     public DFR3MappingResource() {
     }
 
-    // Getter and Setter for hazardType
+    // Getter for type with fallback to dataType if type is not set
     public Type getType() {
-        return type;
+        if (type != null) {
+            return type;
+        }
+        return Type.fromString(mappingType);
+    }
+
```

Comment:
So, how can I update that a workflow is finalized? I didn't see any endpoint or something that handles this. Can we address that in a separate issue?